segmentstore/server/src/main/java/io/pravega/segmentstore/server/SegmentStoreMetrics.java

```diff
@@ -107,13 +109,29 @@ public void close() {
         public ThreadPool(ScheduledExecutorService executor, ScheduledExecutorService storageExecutor) {
             this.executor = Preconditions.checkNotNull(executor, "executor");
             this.storageExecutor = Preconditions.checkNotNull(storageExecutor, "storageExecutor");
-            this.queueSize = STATS_LOGGER.createStats(MetricsNames.THREAD_POOL_QUEUE_SIZE);
-            this.activeThreads = STATS_LOGGER.createStats(MetricsNames.THREAD_POOL_ACTIVE_THREADS);
-            this.storageQueueSize = STATS_LOGGER.createStats(MetricsNames.STORAGE_THREAD_POOL_QUEUE_SIZE);
-            this.storageActiveThreads = STATS_LOGGER.createStats(MetricsNames.STORAGE_THREAD_POOL_ACTIVE_THREADS);
+            queueSize = STATS_LOGGER.createStats(MetricsNames.THREAD_POOL_QUEUE_SIZE);
+            activeThreads = STATS_LOGGER.createStats(MetricsNames.THREAD_POOL_ACTIVE_THREADS);
+            storageQueueSize = STATS_LOGGER.createStats(MetricsNames.STORAGE_THREAD_POOL_QUEUE_SIZE);
+            storageActiveThreads = STATS_LOGGER.createStats(MetricsNames.STORAGE_THREAD_POOL_ACTIVE_THREADS);
             this.reporter = executor.scheduleWithFixedDelay(this::report, 1000, 1000, TimeUnit.MILLISECONDS);
         }
 
+        public OpStatsData getQueueSize() {
```

Comment:
Could you add a `@Getter` annotation in the fields as well, it would be a smaller change. Also, these seem to be needed just for testing, so please add also the `@VisibleForTesting` annotation.

Reply:
Even more, I wonder why in the test you couldn't use the `MetricRegistryUtils` to look for the JVM metric being published instead of adding all these getters.

Reply:
I need to use `Thread.sleep(2000)` for the metric to publish based on the scheduler in `ThreadPool`. I will take out the getters for the stat and use `MetricRegistryUtils` as shown in `SegmentStoreMetricsTests.java`